test/scale/run_test.py

```diff
@@ -0,0 +1,372 @@
+            "kubectl get pods"
+            " -nkube-system"
+            " --selector=app=csi-azurelustre-controller"
+            " --no-headers"
+            " | awk '{print $1}'",
+            need_stdout=True
+        )
+        controller_pods = controller_pods.strip('\n').split('\n')
+        logger.info(f"got controller pods {controller_pods}")
+                
+        node_pods = self.run_command(
+            "kubectl get pods"
+            " -nkube-system"
+            " --selector=app=csi-azurelustre-node"
+            " --no-headers"
+            " | awk '{print $1}'",
+            need_stdout=True
+        )
+        node_pods = node_pods.strip('\n').split('\n')
+        logger.info(f"got nod pods {node_pods}")
+
+        for pod in chain(controller_pods, node_pods):
+            logger.info(f"collecting pod {pod}")
+            self.run_command(f"kubectl logs {pod}"
+                             f" -nkube-system"
+                             f" -cazurelustre"
+                             f" >{self._csi_log_path}/{pod}")
+
+    
+    def parse_result_from_log(self):
+        logger.info("parsing log file for perf result")
+        for log_file in os.listdir(self._csi_log_path):
+            logger.info(f"parsing {log_file}")
+            log_path = os.path.join(self._csi_log_path, log_file)
+            if os.path.isfile(log_path):
+                with open(log_path, "r") as f:
+                    self._perf_result.parse_log_file(f)
+
+    def write_result(self):
+        result_file_path = os.path.join(self._result_path,
+                                        f"result_{self._provisioning_type}"
+                                        f"_{self._current_scale}")
+        perf_result = json.dumps(self._perf_result.get_result())
+        logger.info(f"perf result for scale {self._current_scale}:")
+        logger.info(perf_result)
+        with open(result_file_path, "w") as f:
+            f.write(perf_result)
+        logger.info(f"result have been wrote to {result_file_path}")
+
+    def clean_up(self):
+        logger.info("cleaning")
+        self.delete_workload()
+        if os.path.exists(self._csi_log_path):
+            logger.info("deleting log folder")
+            shutil.rmtree(self._csi_log_path)
+        if os.path.exists(self._generated_workload_yaml):
+            logger.info("deleting tmp yaml file")
+            os.remove(self._generated_workload_yaml)
+
+    def run(self):        
+        for scale in self._scales:
+            logger.info(f"run scale test {scale}")
+            self.configurations["scale"] = str(scale)
+            try:
+                self.setup(scale)
+                self.generate_workload_yaml()
+                self.deploy_workload()
+                time.sleep(10)
+                self.delete_workload()
+                self.collection_logs()
+                self.parse_result_from_log()
+                self.write_result()
+            finally:
```

Comment:
As we discussed, we need one more step to judge the perf results, fail the case if avg of latency_seconds is below certain num or any single latency_seconds is below certain num.